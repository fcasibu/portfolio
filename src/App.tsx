import * as React from 'react';
import styled from 'styled-components';
import { GlobalStyle } from './GlobalStyles';
import { FiCode } from 'react-icons/fi';
import { useModal } from './hooks/useModal';
import Header from './components/Header';
import Showcase from './components/Showcase';
import Projects from './components/Projects';
import AboutModal from './components/AboutModal';

const Wrapper = styled.div`
  width: 100%;
  max-width: 600px;
  margin: 1rem auto;
`;

const SourceCode = styled.a`
  position: fixed;
  right: 15px;
  bottom: 0;
  font-size: 2rem;
`;

function App() {
  const projectsRef = React.useRef<HTMLDivElement | null>(null);
  const aboutRef = React.useRef<HTMLDivElement | null>(null);
  const backdropRef = React.useRef<HTMLDivElement | null>(null);
  const { isOpen, open, close } = useModal();

  const scrollToProjects = () => {
    if (projectsRef.current) {
      projectsRef.current.scrollIntoView({
        behavior: 'smooth'
      });
    }
  };

  return (
    <>
      <GlobalStyle />
      <div ref={backdropRef}></div>
      <div ref={aboutRef}></div>
      <Header scrollToProjects={scrollToProjects} open={open} />
      {isOpen && (
        <AboutModal
          backdropRef={backdropRef.current}
          aboutRef={aboutRef.current}
          close={close}
        />
      )}
      <Wrapper>
        <Showcase scrollToProjects={scrollToProjects} />
        <Projects ref={projectsRef} />
      </Wrapper>
      <SourceCode
        href="https://github.com/nevz9/portfolio/"
        target="_blank"
        rel="noreferrer noopener"
      >
        <FiCode />
      </SourceCode>
    </>
  );
}

export default App;
