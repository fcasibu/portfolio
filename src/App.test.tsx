import { render, screen, within } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('App', () => {
  it('should render the whole app and match te snapshot', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
  it('should call scrollIntoView', () => {
    const mockFn = jest.fn();
    window.HTMLElement.prototype.scrollIntoView = mockFn;

    render(<App />);

    const nav = screen.getByRole('navigation');
    const buttonNav = within(nav).getByText(/projects/i);
    const button = screen.getByRole('button', { name: /view my projects/i });
    userEvent.click(button);
    userEvent.click(buttonNav);
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
