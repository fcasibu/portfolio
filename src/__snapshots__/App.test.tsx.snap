// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the whole app and match te snapshot 1`] = `
<div>
  <div />
  <div />
  <header
    class="sc-bczRLJ hmKztP"
  >
    <nav
      class="sc-gsnTZi cwkucC"
    >
      <span
        class="sc-eCYdqJ kaXuTp"
      >
        nevz
      </span>
      <ul
        class="sc-dkzDqf elkwRH"
      >
        <li
          class="sc-hKMtZM ljcdV"
        >
          Projects
        </li>
        <li
          class="sc-hKMtZM ljcdV"
        >
          About
        </li>
      </ul>
      <button
        class="sc-jSMfEi isVGqD"
      >
        <span>
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Sun
            </title>
            <circle
              cx="12"
              cy="12"
              r="5"
            />
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </svg>
        </span>
      </button>
    </nav>
  </header>
  <div
    class="sc-llJcti gYJzCN"
  >
    <div
      class="sc-iBkjds ekFYN"
    >
      <h1
        class="sc-ftvSup eFeZbx"
      >
        Hi, I'm Francis
      </h1>
      <p
        class="sc-papXJ hmECiM"
      >
        I'm a full stack web developer
      </p>
      <button
        class="sc-gKXOVf jzToMZ"
      >
        View my projects
      </button>
    </div>
    <div
      class="sc-jqUVSM bvSxzK"
    >
      <h2
        class="sc-kDDrLX kHqMnP"
      >
        Projects
      </h2>
      <div
        class="sc-iqcoie ilpSpH"
      >
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD ksUUkG"
            title="Battleship"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="battleship.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Single player battleship game, where the player and computer battles out who sinks the most ships first
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="html.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="css.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="javascript.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="jest.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="webpack.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://nevz9.github.io/battleship/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/battleship"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD dRYaYi"
            title="Fakegram"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="fakegram.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Instagram clone, with features including posting, following, liking, authentication, etc
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="react.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="firebase.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="css.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://nevz9.github.io/fakegram-react/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/fakegram-react"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD kNoOqd"
            title="Photo Tagging Application"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="photo-tagging.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Photo Tagging Application. Three maps. Three Characters. Find all three as fast as you can!
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="react.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="firebase.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="jest.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="css.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://photo-tagging-app-852b2.web.app/#/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/photo-tagging-app"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD hMqVrS"
            title="Shopping Cart"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="shopping-cart.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Shopping Cart. First time implementing React Router and imitating the logic of shoping and adding to a cart
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="react.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="css.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="jest.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://nevz9.github.io/shopping-cart/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/shopping-cart"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD gnNEGX"
            title="Todo App"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="todo.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Todo App. First big project to deal with CRUD operations and using local storage for data persistence
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="html.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="css.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="javascript.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="webpack.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://nevz9.github.io/todo-list/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/todo-list"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-crXcEl bQIyAE"
        >
          <div
            class="sc-breuTD jNSsFw"
            title="Personal Portfolio"
          >
            <img
              class="sc-ksZaOG cNWNAQ"
              src="portfolio.png"
            />
          </div>
          <p
            class="sc-evZas jFFfFl"
          >
            Personal Portfolio. Using styled components and typescript for the first time
          </p>
          <ul
            class="sc-fnykZs bjHzAK"
          >
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="react.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="typescript.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="styled.svg"
              />
            </li>
            <li
              class="sc-fEOsli"
            >
              <img
                alt=""
                class="sc-bjUoiL fHxHZS"
                src="jest.svg"
              />
            </li>
          </ul>
          <ul
            class="sc-hAZoDl dPTzkH"
          >
            <li>
              <a
                href="https://franciscasibu.codes/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/nevz9/portfolio/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <a
    class="sc-iIPllB hSLeFG"
    href="https://github.com/nevz9/portfolio/"
    rel="noreferrer noopener"
    target="_blank"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="16 18 22 12 16 6"
      />
      <polyline
        points="8 6 2 12 8 18"
      />
    </svg>
  </a>
</div>
`;
